src/nerpa_pipeline/handle_rban.py

```diff
@@ -109,43 +166,108 @@ def putative_backbones(G, min_nodes=2):
+    with open(out_file_rban_initial, 'w') as out:
+        json.dump(rban_record, out)
 
 
 def process_single_record(log, rban_record, recognized_monomers, backbone_bond_types,
-                          hybrid_monomers_dict, na=UNDEFINED_NAME, min_recognized_nodes=2):
-    G = build_nx_graph(rban_record, backbone_bond_types, recognized_monomers)
+                          hybrid_monomers_dict, main_out_dir, na=UNDEFINED_NAME, min_recognized_nodes=2):
+    '''
+    build_nx_graphs creates a networkx DiGraph from a monomer graph in rBAN format.
+    In doing so all edges which are not of backbone_bond_types are removed.
+    Also, by default all edges incident to lipid monomers are removed
+    '''
+
+    def add_chirality(graph):
+        try:
+            chirality = handle_monomers.get_monomers_chirality(rban_record)
+            for i, d in chirality.items():
+                graph.nodes[i]['isD'] = d
+        except Exception as e:
+            log.warning(f'Structure "{rban_record["id"]}": unexpected error while determining stereoisomeric configuration '
+                        f'of monomers. Stereoisomeric configuration will be omitted.')
+
+    def add_hybrid_monomers(graph):
+        for i, name in hybrid_monomers_dict.items():
+            G.nodes[i]['name'] = name
+            G.nodes[i]['isPKHybrid'] = True
+            G.nodes[i]['isIdentified'] = True
+
+    '''
+    Try to recognize monomers chirality and
+    add newly identified monomers from hybrid monomers
+    '''
+
     structure_id = rban_record['id']
+    print(f'Processing {structure_id}')
+    G = build_nx_graph(rban_record, backbone_bond_types, recognized_monomers)
+    add_chirality(G)
+    add_hybrid_monomers(G)
 
+    write_rban_record(rban_record, main_out_dir)  # output rban_record for Louis feature
+    '''
+    # I used to need these nerpa processed monomer graphs at some point but now I work directly from rBAN output
+    # Nevertheless, sometime in the future I might need this newly identified monomers or chirality 
     try:
-        chirality = handle_monomers.get_monomers_chirality(rban_record)
-        for i, d in chirality.items():
-            G.nodes[i]['isD'] = d
-    except Exception as e:
-        log.warning(f'Structure "{rban_record["id"]}": unexpected error while determining stereoisomeric configuration '
-                    f'of monomers. Stereoisomeric configuration will be omitted.')
-
-    for i, name in hybrid_monomers_dict.items():
-        G.nodes[i]['name'] = name
-        G.nodes[i]['isPKHybrid'] = True
-        G.nodes[i]['isIdentified'] = True
+        # My version of build_nx_graph, which does not remove any edges and keeps some additional information in the nodes
+        my_G = my_build_nx_graph(rban_record, recognized_monomers)
+        add_chirality(my_G)
+        add_hybrid_monomers(my_G)
+        # I output the initial monomer graph in networkx format to use it later in Louis feature
```

Comment:
both "I" and "Louis feature" are not informative for a third-party reader of these comments